### PR TITLE
module bam2fastx_bam2fastq

### PR DESCRIPTION
**Pull request by muffato**
Cleanup time !
This PR is to bring in a new module developed _more than a year ago_. It's a very old branch and therefore doesn't have nf-test, but it's still perfectly functional. All I had to do was move files around (modules were directly under `modules/` at the time) and update from master.

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/nf-core/bam2fastx/bam2fastq/test.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- name: bam2fastx bam2fastq test_bam2fastx_bam2fastq
+  command: nextflow run ./tests/modules/nf-core/bam2fastx/bam2fastq -entry test_bam2fastx_bam2fastq -c ./tests/config/nextflow.config  -c ./tests/modules/nf-core/bam2fastx/bam2fastq/nextflow.config
+  tags:
+    - bam2fastx
+    - bam2fastx/bam2fastq
+  files:
+    - path: output/bam2fastx/test.fastq.gz
+      md5sum: f0329f9ea12324c6631ac9f3712f2ca5
```
Comment (FloWuenne) [inline tests/modules/nf-core/bam2fastx/bam2fastq/test.yml:8]:
```suggestion
- name: bam2fastx bam2fastq test_bam2fastx_bam2fastq
  command: nextflow run ./tests/modules/nf-core/bam2fastx/bam2fastq -entry test_bam2fastx_bam2fastq -c ./tests/config/nextflow.config
  tags:
    - bam2fastx/bam2fastq
    - bam2fastx
  files:
    - path: output/bam2fastx/test.fastq.gz
      md5sum: f0329f9ea12324c6631ac9f3712f2ca5
    - path: output/bam2fastx/versions.yml
```

**Review by FloWuenne — COMMENTED** [review]
I have added a bunch of suggestions that should fix some linting and testing errors from old code. Hope this helps 👍🏻 !

**Comment by muffato** [comment]
Thank you very much @FloWuenne ! I hadn't noticed that pytest was actually not running (because the paths were wrong).
It's all sorted now, and all 3 tests (Conda/Singularity/Docker) now run and pass, as well as the linting ones.

File: modules/nf-core/bam2fastx/bam2fastq/meta.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - fastq:
+      type: file
+      description: Gzipped FASTQ file
+      pattern: "*.fastq.gz"
+
+authors:
+  - "@priyanka-surana"
```
Comment (FloWuenne) [inline modules/nf-core/bam2fastx/bam2fastq/meta.yml:46]: Maybe add yourself as maintainer @muffato ?

**Review by FloWuenne — APPROVED** [review]
One last comment to maybe add @muffato as author / maintainer, otherwise LGTM!

**Pull request merged**